app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -30,13 +30,22 @@ def fetch_middleware_servers
             @eaps << eap
             server = parse_middleware_server(eap)
 
-            machine_id = @ems.machine_id(eap.feed)
-            host_instance = find_host_by_bios_uuid(machine_id) ||
-                            find_host_by_bios_uuid(alternate_machine_id(machine_id))
-
-            if host_instance
-              server[:lives_on_id] = host_instance.id
-              server[:lives_on_type] = host_instance.type
+            if server.fetch_path(:properties, 'In Container') == 'true'
+              container_id = @ems.container_id(eap.feed)
```

Comment:
Looks like travis is failing on exactly this case [ref](https://travis-ci.org/ManageIQ/manageiq/jobs/218504097#L642)
```
Failure/Error: backing_ref = 'docker://' + container_id
     
     TypeError:
       no implicit conversion of nil into String
```